### ISSUE TITLE
[rewrite] Webhook Support, got ip banned during high traffic

### ISSUE THREAD

**Issue by NNTin** [Rapptz/discord.py#1406]
2 and a half hour ago Discord went down. As a result many people were tweeting at [@discordapp](https://twitter.com/discordapp).  
My bot mirrors the tweets from discordapp to a text channel: https://i.imgur.com/UPe2ZOE.gifv  

During the ordeal by bot was ip banned.

The bot is deployed on Heroku. Here is the Heroku log: https://i.imgur.com/KwzoKqv.png  
The error message is an html web page: https://i.imgur.com/SHVSEi6.png

Not sure why there is a "please enable cookies" message. I assume the bot was ip banned due to going above the rate limit.

I looked at the webhook code and it seem rate limiting is handled. Not sure what exactly caused my bot to be ip banned. It happened during high traffic. My other webhooks that are also following twitter users are working fine (but they didn't tweet as much as discordapp during peak time).

edit: thought webhook didn't handle rate limits, it does. Rephrased my post.

**Comment by Rapptz** [comment]
This requires more information to be actionable. Not only is the original accusation wrong (I do handle rate limits) but despite being incorrect the accusation that you got banned still somehow decided to be put on me.

**Comment by Harmon758** [comment]
You should contact [Discord Support](https://support.discordapp.com/hc/en-us) and figure out why it was banned before placing the blame on discord.py.

**Comment by NNTin** [comment]
I contacted Discord Support.  "A bot broke our Developer ToS by implementing a feature which spammed our API so we blocked the IP Address that it was tied to."

My webhook bot sends the tweet to discord pretty much the moment the tweet is created. But when my bot was banned it was lagging 5 minutes behind due to rate limiting: https://i.imgur.com/hi25YmA.png

I suspect the bot was rate limited. Discord.py was waiting and handled the rate limit properly. But a new tweet came in and immediately another webhook was sent. Correct me if I am wrong. I tried looking into the code but I have struggles reaffirming my thoughts.
